lib/concurrent/edge/future.rb

```diff
@@ -43,38 +44,66 @@ def future(default_executor = :io, &task)
 
       alias_method :async, :future
 
-      # Constructs new Future which will be completed after block is evaluated on executor. Evaluation is delayed until
-      # requested by `#wait`, `#value`, `#value!`, etc.
-      # @return [Delay]
-      def delay(default_executor = :io, &task)
-        Delay.new(default_executor).event.chain(&task)
+      # Constructs new Future which will evaluate to the block after
+      # requested by calling `#wait`, `#value`, `#value!`, etc. on it or on any of the chained futures.
+      # @return [Future]
+      def delay(*args, &task)
+        delay_on :io, *args, &task
+      end
+
+      # @return [Future]
+      def delay_on(default_executor, *args, &task)
+        Delay.new(default_executor, *args).future.then(&task)
       end
 
       # Schedules the block to be executed on executor in given intended_time.
       # @return [Future]
-      def schedule(intended_time, default_executor = :io, &task)
-        ScheduledPromise.new(intended_time, default_executor).future.chain(&task)
+      def schedule(intended_time, *args, &task)
```

Comment:
Can we list what types can be used with IntendedTime? This may have some implications if used with something like ActiveSupport, for example.